src/pages/[...slug].tsx

```diff
@@ -62,41 +68,35 @@ const componentsMapping = {
   tutorial: tutorialsComponents,
 } as const
 
-interface Params extends ParsedUrlQuery {
-  slug: string[]
-}
-
-interface Props {
-  mdxSource: MDXRemoteSerializeResult
-}
-
-export const getStaticPaths: GetStaticPaths = ({ locales }) => {
+export const getStaticPaths = (({ locales }) => {
   const contentFiles = getContent("/")
 
-  let paths: StaticPaths = []
-
   // Generate page paths for each supported locale
-  for (const locale of locales!) {
-    contentFiles.map((file) => {
-      paths.push({
-        params: {
-          // Splitting nested paths to generate proper slug
-          slug: file.slug.split("/").slice(1),
-        },
-        locale,
-      })
-    })
-  }
+  const paths = locales!.flatMap((locale) =>
+    contentFiles.map((file) => ({
+      params: {
+        // Splitting nested paths to generate proper slug
+        slug: file.slug.split("/").slice(1),
+      },
+      locale,
+    }))
+  )
 
   return {
     paths,
     fallback: false,
   }
-}
+}) satisfies GetStaticPaths<{ slug: string[] }>
+
+type Props = Omit<
+  Parameters<LayoutMappingType[keyof LayoutMappingType]>[0],
+  "children"
+> &
+  SSRConfig & {
+    mdxSource: MDXRemoteSerializeResult
+  }
```

Comment:
This type is for `GetStaticProps` to thoroughly define the props object returned.